tests/utils/factories.py

```diff
@@ -320,12 +486,48 @@ class Meta:
 
     uids = factory.SubFactory(IDDataFactory)
     name = factory.Faker('color_name')
-    tags = []
-    notes = None
+    tags = factory.List([factory.Faker('color_name'), factory.Faker('color_name')])
+    notes = factory.Faker('catch_phrase')
     process = factory.SubFactory(LinkByUIDFactory)
-    sample_type = 'experimental'
-    spec = None
-    file_links = []
+    sample_type = factory.Faker("random_element", elements=list(SampleType))
+    spec = factory.SubFactory(LinkByUIDFactory)
+    file_links = factory.List([factory.SubFactory(FileLinkFactory)])
+
+
+class LinkByUIDDataFactory(factory.DictFactory):
+    id = LinkByUIDFactory.id
+    scope = LinkByUIDFactory.scope
+    type = 'link_by_uid'
+
+
+class FileLinkDataFactory(factory.DictFactory):
+    url = FileLinkFactory.url
+    filename = FileLinkFactory.filename
+    type = 'file_link'
+
+
+class MaterialSpecDataFactory(factory.DictFactory):
+    uids = MaterialSpecFactory.uids
+    name = MaterialSpecFactory.name
+    tags = MaterialSpecFactory.tags
+    notes = MaterialSpecFactory.notes
+    process = factory.SubFactory(LinkByUIDDataFactory)
+    file_links = factory.List([factory.SubFactory(FileLinkDataFactory)])
+    template = factory.SubFactory(LinkByUIDDataFactory)
+    properties = []  # TODO make a PropertiesDataFactory
+    type = 'material_spec'
```

Comment:
There are likely cleaner patterns here, particularly using [make_factory](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.make_factory), but I didn't want to get into too much magic until we had a plan for moving forward leveraging the API.